### PR TITLE
Three API endpoints for getting information on projects, submissions,…

### PR DESCRIPTION
**Pull request by Tsintsir**
… and classes relevant in some way to a specified user.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -324,3 +324,21 @@ def remove_contributor(username, submission_name, removed_username):
 def get_contributors(username, submission_name):
     dbw = DatabaseWrapper(GITOLITE_ADMIN_PATH, DATABASE_PORT)
     return dbw.get_contributors(username, submission_name)
+
+
+@app.route('/<username>/projects', methods=['GET'])
+def get_projects(username):
+    dbw = DatabaseWrapper(GITOLITE_ADMIN_PATH, DATABASE_PORT)
+    return dbw.get_projects(username)
```
Comment (shawkinsl) [inline src/app.py]: These need to be json format in order to work in flask ( `return jsonify({"projects": dbw.get_projects(username)})`

File: src/db.py [hunk 2]
```diff
@@ -263,3 +263,21 @@ def get_contributors(self, username, submission_name):
...
+
+    def get_projects(self, username):
+        project_db = self.mongo.gitsubmit.projects
+        resultCursor = project_db.find({"owner": username})
+        return resultCursor.toArray()
+
+
+    def get_submissions(self, username):
+        submission_db = self.mongo.gitsubmit.submissions
+        resultCursor = submission_db.find( { "$or": [ {"owner": username}, {"contributors": username} ] })
```
Comment (shawkinsl) [inline src/db.py]: Did you test out this query? I think contributors needs to be an `"$in"` query, but I remember mongo sometimes doing things correctly automatically

Comment (shawkinsl) [inline src/db.py]: Nevermind, I'm wrong: http://stackoverflow.com/questions/18148166/find-document-with-array-that-contains-a-specific-value

File: src/db.py [hunk 3]
```diff
@@ -263,3 +263,21 @@ def get_contributors(self, username, submission_name):
         if submission_doc is None:
             raise SubmissionDoesNotExistError(str(username) + ": " + str(submission_name))
         return submission_doc["contributors"]
+
+
+    def get_projects(self, username):
+        project_db = self.mongo.gitsubmit.projects
+        resultCursor = project_db.find({"owner": username})
```
Comment (shawkinsl) [inline src/db.py]: would prefer snake_case for variables

File: src/db.py [hunk 4]
```diff
@@ -263,3 +263,21 @@ def get_contributors(self, username, submission_name):
...
+        return resultCursor.toArray()
```
Comment (shawkinsl) [inline src/db.py]: Oh, one other thing: datetimes in projects need to be converted to a string that the frontend will understand (see https://github.com/gitsubmit/gitsubmit-backend/blob/master/src/db.py#L105 )

**Comment by shawkinsl** [comment]
+1 and will merge once comments are addressed

File: src/db.py (hunk 4 above)
Comment (Tsintsir) [inline src/db.py]: Forgot about those, will be an easy fix. Thanks!

**Comment by Tsintsir** [comment]
This should be good!

**Comment by shawkinsl** [comment]
I'm going to make a few minor changes here, then merge

**Pull request merged**